Geometry_Engine/Compute/BooleanIntersection.cs

```diff
@@ -378,19 +416,22 @@ public static List<PolyCurve> BooleanIntersection(this IEnumerable<ICurve> regio
         /***          Private methods                    ***/
         /***************************************************/
 
-        private static Boolean IsSimilarSegment(this ICurve curve, ICurve refCurve, double tolerance = Tolerance.Distance)
+        [Description("Checks if two segments are similar by comparing the start and end points (allowing for reversed order, i.e. start against end) as well as the mid points.")]
+        private static bool IsSimilarSegment(this ICurve curve, ICurve refCurve, double tolerance = Tolerance.Distance)
         {
             double sqTol = tolerance * tolerance;
             Point sp = curve.IStartPoint();
             Point rsp = refCurve.IStartPoint();
-            Point mp = curve.IPointAtParameter(0.5);
-            Point rmp = refCurve.IPointAtParameter(0.5);
             Point ep = curve.IEndPoint();
             Point rep = refCurve.IEndPoint();
-
-            return mp.SquareDistance(rmp) <= sqTol && 
-                   ((sp.SquareDistance(rsp) <= sqTol && ep.SquareDistance(rep) <= sqTol) ||
-                   (sp.SquareDistance(rep) <= sqTol && ep.SquareDistance(rsp) <= sqTol));
+            if ((sp.SquareDistance(rsp) <= sqTol && ep.SquareDistance(rep) <= sqTol) ||
+                   (sp.SquareDistance(rep) <= sqTol && ep.SquareDistance(rsp) <= sqTol))
+            {
+                Point mp = curve.IPointAtParameter(0.5);
```

Comment:
This is not intended to have any functional change - The midpoint where compared on main as well (see first statement on main `mp.SquareDistance(rmp) <= sqTol` ). 

What this change is doing is to first check if the starts/ends are matching before the midpoint is extracted. Reason for this is that PointAtParameter is slower than start and end point, hence wanted to only extract it if necessary.

The way I read it (and intention) is main:

- Extract start, mid and end points
- Compare midpoints, if same, compare start and end points (allowing flips).

Branch:

- Extract start and end points
- Compare start and endpoints (allowing flips)
- If same extract mid point
- Compare midpoint

Could agree with you that we maybe should compare more points, but would prefer to do that on a separate PR, as the intention here is just to change the execution order, not the behaviour. If you still do not agree that the functionality should be the same, then I am happy to update (in case I have missed something 😄 )

Reply:
No no, you are right, I misread the code. No functional change is always welcome, particularly when the code works well (and I am pretty sure about that, remember @LMarkowski's struggle on that particular one 😃 😃 😃)